### PR TITLE
Plugins

### PR DESCRIPTION
**Pull request by danderson00**
Adds the ability for known plugins (azure-mobile-apps-files is currently the only known plugin) to be loaded automatically.

@mamaso

### CODE REVIEW

**Comment by mamaso** [comment]
I like the automatic loading if module exists, pretty slick.  Good to log so that there's no silent plugin load surprises.

Do we want to add a mechanism for community plugins as well?  Could do something like a 'plugins' attribute on the configuration, which either has the plugin object or a string to require.
```js
plugins: {
  community-plugin: stringOrObject
}
```

Probably unnecessary as we already have the routers for custom mounting.

File: src/express/knownPlugins.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+        try {
+            var pluginModule = require(name);
+            try {
+                plugins.push(pluginModule(configuration, logger));
+                logger.info('Loaded plugin ' + name);
+            } catch (ex) {
+                logger.error('Found plugin ' + name + ' but failed to load: ', ex);
+            }
+        } catch(ex) {
+            // ignore module loading errors, probably means the module has not been installed
```
Comment (mamaso) [inline src/express/knownPlugins.js]: still might want to log at a 'silly' level if the module loading fails, could be useful info for a confused user

Comment (danderson00) [inline src/express/knownPlugins.js]: Yeh, probably not a bad idea. It should really only ever fail if the module hasn't been installed with npm, but there could be other reasons as well. Will add.

**Comment by danderson00** [comment]
The mechanism currently just loads the plugin in exactly the same way as calling mobileApp.use, so it's trivial to load other custom "plugins", but we could make this simpler in future if it ends up being used.

**Comment by danderson00** [comment]
Resolves https://github.com/Azure/azure-mobile-apps-node/issues/431

**Pull request merged**